client/client.go

```diff
@@ -131,14 +132,34 @@ func (a *ApiClient) Request(code AuthToken) *ApiRequest {
 type ApiRequest struct {
 	api *ApiClient
 	// authorization code to use for requests
-	auth AuthToken
+	auth   AuthToken
+	revoke func() (AuthToken, error)
 }
 
 func (ar *ApiRequest) Do(req *http.Request) (*http.Response, error) {
 	if req.Header.Get("Authorization") == "" {
 		req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", ar.auth))
 	}
-	return ar.api.Do(req)
+	r, err := ar.api.Do(req)
+	if r != nil && r.StatusCode == http.StatusUnauthorized {
+		// invalid JWT-token
+		log.Info("Device unauthorized; attempting reauthorization")
+		if jwt, e := ar.revoke(); e == nil {
+			ar.auth = jwt
+			if req.GetBody != nil {
+				// if body present: retrieve body
+				if body, e := req.GetBody(); e == nil {
+					req.Body = body
+				}
+			}
+			// retry API request with new JWT token
+			req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", ar.auth))
+			r, err = ar.api.Do(req)
+		} else {
+			log.Info("Reauthorization failed")
```

Comment:
We could do `log.Infof("Reauthorization failed with error: %s", err.Error())` to get some more data why this error happened.